File: Phase01/src/InvertedIndex.java
import java.util.HashMap;
import java.util.HashSet;
import java.util.Set;

public class InvertedIndex {
    private static final HashMap<String, Set<String>> invertedIndex = new HashMap<>();

    public HashMap<String, Set<String>> getInvertedIndex() {
        return invertedIndex;
    }

    public void indexDocument(String document, String docID) {
        String[] words;

        words = document.split(" ");
        for (String word : words)
            if (word.length() > 1)
                add_word(word, docID);
    }

    private void add_word(String word, String docID) {
        boolean wordExists = invertedIndex.containsKey(word);
        Set<String> list;

        if (wordExists) list = invertedIndex.get(word);
        else list = new HashSet<>();

        list.add(docID);
        invertedIndex.put(word, list);
    }
}

File: Phase05/SampleLibrary/SampleLibrary/InvertedIndex.cs
namespace SampleLibrary;

public class InvertedIndex
{
    public Dictionary<string, HashSet<string>> Map { get; } = new();

    public void indexDocument(string document, string docId)
    {
        document.Split(" ").Where(word => word.Length > 1).ToList().ForEach(word => AddWordToMap(word, docId));
    }

    private void AddWordToMap(string word, string docId)
    {
        HashSet<string> docIdSet;

        if (Map.ContainsKey(word))
            docIdSet = Map[word];
        else
            docIdSet = new HashSet<string>();

        docIdSet.Add(docId);
        Map[word] = docIdSet;
    }
}